src/eliminate_contiguous.cpp

```diff
@@ -41,6 +41,14 @@ static bool try_compute_shape(instruction_ref ins,
                               const std::vector<shape>& inputs,
                               const std::vector<module_ref>& mods)
 {
+    // compute_shape of gpu::contiguous always returns a standard shape
+    // this can propagate false `true` values for non standard shapes in recursive try_compute_shape
+    // calls
+    if(ins->name() == "gpu::contiguous" &&
```

Comment:
This is checking the wrong name. This should be using `op_name` that is a member variable of the `eliminate_contiguous` class.

Reply:
The main reason I was using `ins->name()` instead of `op_name` here, because recursive `try_compute_shape` calls can process `gpu::contiguous` and that's where the issue comes from in my opinion. But I can aggree that this doesn't apply to the idea what you mentioned above:
> We should remove contiguous when the inputs are not standard shape as well if the outputs can handle the non-standard shape.
I will have another iteration on this. First I will start with a unit test, maybe that can highlight the issue better.